evalml/pipelines/components/transformers/preprocessing/delayed_feature_transformer.py

```diff
@@ -16,10 +16,11 @@ class DelayedFeatureTransformer(Transformer):
     hyperparameter_ranges = {}
     needs_fitting = False
 
-    def __init__(self, max_delay=2, delay_features=True, delay_target=True, gap=1, random_seed=0, **kwargs):
+    def __init__(self, date_index=None, max_delay=2, delay_features=True, delay_target=True, gap=1, random_seed=0, **kwargs):
```

Comment:
Mainly added for consistency across time series estimators/transformers

Reply:
Yeah, makes sense, I think every time series component needs the time series configuration.